task-generator/remote/main.go

```diff
@@ -102,14 +102,16 @@ func convertToSsh(task *tektonapi.Task, taskVersion string) {
 	// The images produced in multi-platform builds need to have unique tags in order
 	// to prevent them from getting garbage collected before generating the image index.
 	// We can simplify this process, preventing the need for users to manually specify
-	// the image by auto-appending the architecture from the PLATFORM parameter. For
-	// example, this will append -arm64 if PLATFORM is linux/arm64 if not present. Since
-	// we cannot modify the parameter itself, this replacement needs to happen in any task
-	// step where the IMAGE parameter is used.
-	// If a user defines the IMAGE parameter with an -arm64 suffix, the arm64 suffix will
-	// not be appended again based on the PLATFORM.
-	adjustRemoteImage := `if [[ "${IMAGE##*-}" != "${PLATFORM##*/}" ]]; then
-  export IMAGE="${IMAGE}-${PLATFORM##*/}"
+	// the image by auto-appending the a sanitized PLATFORM parameter. For example, this
+	// will append linux-arm64 if PLATFORM is linux/arm64 and IMAGE_APPEND_PLATFORM is true.
+	// Many special characters are not allowed in tags so we will replace anything that
+	// isn't alphanumeric with a "-" to be safe. Since we cannot modify the parameter itself,
```

Comment:
Should we fail the pipeline if users specify invalid tag? Then we don't have to check / replace in every task.

Reply:
The users are not providing a tag. They are providing the PLATFORM and we are sanitizing that variable so that we can use it as a tag suffix.